retail/interactive-tutorial/RetailSearch.Samples/SearchWithNextPageTokenSample.cs

```diff
@@ -0,0 +1,129 @@
+
+        var searchRequest = new SearchRequest()
+        {
+            Placement = defaultSearchPlacement, // Placement is used to identify the Serving Config name
+            VisitorId = "123456", // A unique identifier to track visitors
+            Query = query,
+            PageToken = nextPageToken
+        };
+
+        Console.WriteLine("Search. request:");
+        Console.WriteLine($"Placement: {searchRequest.Placement}");
+        Console.WriteLine($"Query: {searchRequest.Query}");
+        Console.WriteLine($"VisitorId: {searchRequest.VisitorId}");
+        Console.WriteLine($"PageSize: {searchRequest.PageSize}");
+        Console.WriteLine();
+
+        return searchRequest;
+    }
+
+    /// <summary>
+    /// Call the retail search.
+    /// </summary>
+    /// <param name="projectNumber">Current project number.</param>
+    /// <returns></returns>
+    public IEnumerable<SearchResponse> Search(string projectNumber)
+    {
+        // Call with empty next page token:
+        string nextPageToken = "";
+        string query = "Hoodie";
+
+        SearchServiceClient client = SearchServiceClient.Create();
+        SearchRequest searchRequest = GetSearchRequest(query, nextPageToken, projectNumber);
+        IEnumerable<SearchResponse> firstSearchResultPages = client.Search(searchRequest).AsRawResponses();
+        SearchResponse firstPage = firstSearchResultPages.FirstOrDefault();
+
+        if (firstPage is null)
+        {
+            Console.WriteLine("The search operation returned no matching results.");
+        }
+        else
+        {
+            Console.WriteLine("First page search results:");
+            Console.WriteLine($"AttributionToken: {firstPage.AttributionToken},");
+            Console.WriteLine($"NextPageToken: {firstPage.NextPageToken},");
+            Console.WriteLine($"TotalSize: {firstPage.TotalSize},");
+            Console.WriteLine("Items found in first page:");
+
+            foreach (SearchResponse.Types.SearchResult item in firstPage)
+            {
+                Console.WriteLine(item);
+            }
+        }
+
+        // Call with valid next page token:
+        nextPageToken = firstPage.NextPageToken;
+        searchRequest = GetSearchRequest(query, nextPageToken, projectNumber);
+        IEnumerable<SearchResponse>  secondSearchResultPages = client.Search(searchRequest).AsRawResponses();
+        SearchResponse secondPage = secondSearchResultPages.FirstOrDefault();
+
+        if (secondPage is null)
+        {
+            Console.WriteLine("The search operation returned no matching results.");
+        }
+        else
+        {
+            Console.WriteLine("Second page search results:");
+            Console.WriteLine($"AttributionToken: {secondPage.AttributionToken},");
+            Console.WriteLine($"NextPageToken: {secondPage.NextPageToken},");
+            Console.WriteLine($"TotalSize: {secondPage.TotalSize},");
+            Console.WriteLine("Items found in first page:");
```

Comment:
second page

Reply:
This is goinf to be executed several times, not necessarily for the first page only, so I think it's best like this:
```suggestion
            Console.WriteLine("Items found in page:");
```